apps/passport-server/test/generic-issuance/util.ts

```diff
@@ -157,49 +139,76 @@ export async function requestTicketsFromPipeline(
 ): Promise<EdDSATicketPCD[]> {
   const ticketPCDResponse = await requestPollFeed(feedUrl, {
     feedId: feedId,
-    pcd: await signFeedCredentialPayload(
+    pcd: await makeCredential(
       identity,
-      createFeedCredentialPayload(
-        await EmailPCDPackage.serialize(
-          await EmailPCDPackage.prove({
-            privateKey: {
-              value: zupassEddsaPrivateKey,
-              argumentType: ArgumentTypeName.String
-            },
-            id: {
-              value: "email-id",
-              argumentType: ArgumentTypeName.String
-            },
-            emailAddress: {
-              value: email,
-              argumentType: ArgumentTypeName.String
-            },
-            semaphoreId: {
-              value: identity.commitment.toString(),
-              argumentType: ArgumentTypeName.String
-            }
-          })
-        )
-      )
+      PODBOX_CREDENTIAL_REQUEST,
+      email,
+      zupassEddsaPrivateKey
     )
   });
 
   return getTicketsFromFeedResponse(expectedFolder, ticketPCDResponse);
 }
 
 /**
- * TODO: extract this to the `@pcd/passport-interface` package.
+ * Makes a credential for a given email address and Semaphore identity, by
+ * generating a new Email PCD using the provided private key.
+ *
+ * Uses {@link testCredentialCache} to avoid regenerating the same credential
+ * repeately.
  */
-export async function signFeedCredentialPayload(
+export async function makeCredential(
   identity: Identity,
-  payload: FeedCredentialPayload
-): Promise<SerializedPCD<SemaphoreSignaturePCD>> {
+  request: CredentialRequest,
+  email?: string,
+  zupassEddsaPrivateKey?: string
+): Promise<Credential> {
+  if (request.pcdType === "email-pcd") {
+    if (!email || !zupassEddsaPrivateKey) {
+      throw new Error(
+        "Can't create a credential containing an EmailPCD without email address and private key"
+      );
+    }
+    const emailPCD = await proveEmailPCD(
+      email,
+      zupassEddsaPrivateKey,
+      identity
+    );
+    // Credential Manager will need to be able to look up the Email PCD, and use
+    // an identity. We instantiate a PCDCollection here, mirroring the usage on
+    // the client.
+    const credentialManager = new CredentialManager(
+      identity,
+      new PCDCollection([EmailPCDPackage], [emailPCD]),
+      new Map()
+    );
+    return credentialManager.requestCredential(request);
+  } else {
+    // No Email PCD required here
+    const credentialManager = new CredentialManager(
+      identity,
+      new PCDCollection([], []),
+      new Map()
+    );
+    return credentialManager.requestCredential(request);
+  }
+}
+
+/**
+ * Sign a credential payload.
+ * Only use this to generate potentially incorrect credentials, otherwise use
```

Comment:
could you expand on what a `potentially incorrect credential` is?

Reply:
Done